projects/aca-playwright-shared/src/page-objects/pages/search.page.ts

```diff
@@ -44,4 +44,36 @@ export class SearchPage extends BasePage {
   public searchOverlay = new SearchOverlayComponent(this.page);
   public sidenav = new SidenavComponent(this.page);
   public confirmDialogComponent = new AdfConfirmDialogComponent(this.page);
+
+  async searchFilesAndFolders(searchText: string): Promise<void> {
+    await this.acaHeader.searchButton.click();
+    await this.searchInput.searchButton.click();
+    await this.searchInput.checkFilesAndFolders();
+    await this.searchInput.searchFor(searchText);
+    await this.dataTable.progressBarWaitForReload();
+  }
+
+  async searchFiles(fileName: string): Promise<void> {
+    await this.acaHeader.searchButton.click();
+    await this.searchInput.searchButton.click();
+    await this.searchInput.checkOnlyFiles();
+    await this.searchInput.searchFor(fileName);
+    await this.dataTable.progressBarWaitForReload();
+  }
+
+  async searchFolders(folderName: string): Promise<void> {
+    await this.acaHeader.searchButton.click();
+    await this.searchInput.searchButton.click();
+    await this.searchInput.checkOnlyFolders();
+    await this.searchInput.searchFor(folderName);
+    await this.dataTable.progressBarWaitForReload();
+  }
+
+  async searchLibraries(libraryName: string): Promise<void> {
+    await this.acaHeader.searchButton.click();
+    await this.searchInput.searchButton.click();
+    await this.searchInput.checkLibraries();
```

Comment:
Basically each of those function is exactly the same except this line, I think we can make one function out of that with additional param which might be something like `searchType` which can accept values like `files | folders | libraries | filesAndFolders` and then you can use `switch case` to properly call correct action based on  `searchType` value

Reply:
Done